app/src/main/java/com/example/roomies/HomeFragment.java

```diff
@@ -44,6 +72,149 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+        query.include(UserCircle.KEY_CIRCLE);
+        // start an asynchronous call for UserCircle objects that include current user
+        query.findInBackground(new FindCallback<UserCircle>() {
+            @Override
+            public void done(List<UserCircle> userCircles, ParseException e) {
+                // check for errors
+                if (e != null) {
+                    Log.e(TAG, "Issue with getting userCircles", e);
+                    return;
+                }
+
+                // user has not joined a circle
+                if(userCircles.isEmpty()){
+                    Intent i = new Intent(getActivity(), AddCircleActivity.class);
+                    startActivity(i);
+                    getActivity().finish();
+                }
+
+                // save received posts to list and notify adapter of new data
+                circle = userCircles.get(0).getCircle();
+
+                // fill image and text on home screen
+                Glide.with(context).load(circle.getImage().getUrl()).apply(RequestOptions.circleCropTransform()).into(ivCirclePhoto);
+                tvCircleName.setText(circle.getName());
+
+                getAllProfiles(view);
+            }
+        });
+    }
+
+    /**
+     * Get all UserCircles related to current circle
+     */
+    public void getAllProfiles(View view){
+        // find profile of all users in current circle - query UserCircle.class
+        ParseQuery<UserCircle> query = ParseQuery.getQuery(UserCircle.class).whereEqualTo(UserCircle.KEY_CIRCLE, circle);
+        // include data referred by user key
+        query.include(UserCircle.KEY_USER);
+        // start an asynchronous call for UserCircle objects that include current circle
+        query.findInBackground(new FindCallback<UserCircle>() {
+            @Override
+            public void done(List<UserCircle> userCircles, ParseException e) {
+                // check for errors
+                if (e != null) {
+                    Log.e(TAG, "Issue with getting userCircles", e);
+                    return;
+                }
+
+                // user has not joined a circle
+                if(userCircles.isEmpty()){
+                    Intent i = new Intent(getActivity(), AddCircleActivity.class);
+                    startActivity(i);
+                    getActivity().finish();
+                }
+
+                // save received posts to list and notify adapter of new data
+                userCircleList.clear();
+                userCircleList.addAll(userCircles);
+                if(userCircleList.size() > 0){
+                    fillProfileImages(view);
+                }
+            }
+        });
+    }
+
+    /**
+     * Show profile images of all users in circle
+     * @param view
+     */
+    public void fillProfileImages(View view){
+        Log.e(TAG, userCircleList.toString());
+        if(userCircleList.size() > 0 && userCircleList.get(0).getUser().getParseFile("image") != null){
```

Comment:
with the profiles as a list, this can also be optimized via a for-loop

Reply:
I'll add this as a TODO. I might need to change a lot of things with how I format my layout in order to implement this. I'll also look into if it's possible to concatenate variable names in Java. Couldn't find anything as of now.